plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -401,6 +419,28 @@ private AtomicReference<PartitionStatistics> refreshTableStatistics(HiveTableNam
     private static <K, V> V get(LoadingCache<K, V> cache, K key)
     {
         try {
+            V value = cache.getUnchecked(key);
+            checkState(!(value instanceof Optional), "This must not be used for caches with Optional values, as it doesn't implement cacheMissing logic. Use getOptional()");
+            return value;
+        }
+        catch (UncheckedExecutionException e) {
+            throwIfInstanceOf(e.getCause(), TrinoException.class);
+            throw e;
+        }
+    }
+
+    private <K, V> Optional<V> getOptional(LoadingCache<K, Optional<V>> cache, K key)
+    {
+        try {
+            Optional<V> value = cache.getIfPresent(key);
+            @SuppressWarnings("OptionalAssignedToNull")
+            boolean valueIsPresent = value != null;
+            if (valueIsPresent) {
+                if (value.isPresent() || cacheMissing) {
+                    return value;
+                }
+                cache.invalidate(key);
+            }
             return cache.getUnchecked(key);
```

Comment:
Is it necessary call this method if the value is not present?

Reply:
yes, `cache.getIfPresent` didn't load anything yet